extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -353,8 +376,20 @@ export class AmpSelector extends AMP.BaseElement {
     const newSelectedOption = this.options_[this.focusedIndex_];
     newSelectedOption.tabIndex = 0;
     tryFocus(newSelectedOption);
+
+    const focusedOption = this.options_[this.focusedIndex_];
     if (this.kbSelectMode_ == KEYBOARD_SELECT_MODES.SELECT) {
-      this.onOptionPicked_(newSelectedOption);
+      this.onOptionPicked_(focusedOption);
+    }
+  }
+
+  selectionKeyDownHandler_(event) {
+    const keyCode = event.keyCode;
+    if (keyCode == Keycodes.SPACE || keyCode == Keycodes.ENTER) {
+      event.preventDefault();
```

Comment:
move `preventDefault` inside the `if (this.options_.includes(event.target))` otherwise actionable elements inside `option` can be broken ( e.g. `<div option> <button> </div>`)